### DISCUSSION TITLE
Broadcast to current window not working using recent:0

### DISCUSSION THREAD

**Discussion by Weavus** [kovidgoyal/kitty#4405] [Q&A] [answered]
Hi

First of all, thank you so much for the fantastic terminal.

I'm trying to get a couple of keymappings going for using the broadcast kitty and I can get it to broadcast to a single tab fine but I can't seem to make it work for a single window. The keymappings I'm using are as follows

map F1 launch --allow-remote-control kitty +kitten broadcast --match-tab=recent:0
map F2 launch --allow-remote-control kitty +kitten broadcast --match=recent:0

Is there something I'm doing wrong with the F2 mapping to broadcast to all tabs in the current window? When I try to use it I get the broadcast window come up as expected but nothing is broadcast to _any_ windows/tabs open.

Thanks

**Comment by page-down** [comment, marked as answer]
This is because `broadcast --match recent:0` refers to the current active window.
The currently active kitty window is the broadcast input window, which is excluded by default, so it has no effect.

Maybe you want `broadcast --match recent:1`, the last active window.

EDIT:
I just reread your description. You seem to want to broadcast for all kitty windows in all tabs in one OS window? Then the existing functionality cannot do that and you need to write (extend) the current broadcast kitten.

**Comment by Weavus** [comment]
Hi, thanks for the quick reply. 

Yes, I think I was confused as to what a 'window' was in this case thinking it was an OS window rather than an application window. I'll take a look at the kitten to see if I can extend it to do what I want. Thanks for clarifying.
